internal/controller/persistentvolume/persistentvolume.go

```diff
@@ -122,6 +122,40 @@ func checkStaticVolume(pv *corev1.PersistentVolume) (bool, error) {
 	return static, nil
 }
 
+// storeVolumeIDInPV stores the new volumeID in PV object.
+func (r ReconcilePersistentVolume) storeVolumeIDInPV(pv *corev1.PersistentVolume, newVolumeID string) error {
+	// take a lock to avoid concurrent update of configmap
+	if v, ok := pv.Annotations[rbd.PVVolumeHandleLabel]; ok {
+		if v == newVolumeID {
+			return nil
+		}
+	}
+	if pv.Annotations == nil {
+		pv.Annotations = make(map[string]string)
+	}
+	if pv.Labels == nil {
+		pv.Labels = make(map[string]string)
+	}
+	volumeIDExistInAnnotation := false
+	replicationLabelExists := false
+	// check VolumeID exists in annotation and matching
+	if v, ok := pv.Annotations[rbd.PVVolumeHandleLabel]; ok {
+		if v == newVolumeID {
+			volumeIDExistInAnnotation = true
+		}
+	}
+	// check PV label exists
+	if _, ok := pv.Labels[rbd.PVReplicatedLabel]; ok {
+		replicationLabelExists = true
+	}
+	if replicationLabelExists && volumeIDExistInAnnotation {
+		return nil
+	}
+	pv.Labels[rbd.PVReplicatedLabel] = "true"
```

Comment:
I'm not a big fan of `true` strings, maybe some status like `volume-handle-detected`? Just a nit, not required to change as the value is not *that* important.